app/scripts/metamask-controller.js

```diff
@@ -308,6 +309,13 @@ module.exports = class MetamaskController extends EventEmitter {
     }
   }
 
+  setLocked () {
+    this.keyringController.setLocked()
+      .then((data) => {
+        this.platform.reload()
```

Comment:
I would like us to avoid reloading from within the metamask controller and maybe instead emit a locked event and reload in background.js